[11333] Update the splash screen's layout

ankitrox opened the pull request:
## Summary

<!-- Please reference the issue this PR addresses in the following list. -->
Addresses issue:

- #11333

## Relevant technical choices

As per the [discussion](https://fueled.slack.com/archives/C077DB69US2/p1758807507359689) with @techanvil, the clipping of screenshot at the bottom of the viewport is not added in this PR because we need to get clarity over the admin footer where it will be visible anyways causing the vertical scrollbar to appear on the page. Once we finalize over the footer, we can add the change for fix height of the main container.

<!-- Please describe your changes. -->

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [x] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.
- [x] Ensure there are no unexpected significant changes to file sizes.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [x] Add a changelog message to the issue.

techanvil reviewed (changes requested):
Hey @ankitrox, this is off to a good start, but there are a few issues to address. Please see my comments.

Additionally, the margin of the content has a grey background, whereas it should be white as per the design.

<img width="1514" height="843" alt="image" src="https://github.com/user-attachments/assets/656cd148-77b4-4300-bec2-45a1abefb588" />

For reference, here's a screen grab of the Figma design. This shows how the background should be white to the edge of the viewport.

<img width="1130" height="599" alt="image" src="https://github.com/user-attachments/assets/e4b7d355-fcd8-4b0e-9cb4-bd52a7a54e23" />

One more point, the screenshot is not being clipped when it's taller than the page - you can see the scrollbar in my screen grab, which again should not be present in this case.

As per the AC:

> ... for viewports which are shorter than the screenshot image, the intention is to have the screenshot image clipped at the bottom edge of the viewport rather than allowing it to overflow and be scrollable.

ankitrox commented:
Thank you @techanvil for reviewing the PR and adding your feedback.

I've addressed the feedback except the clipping of screenshot at bottom of the viewport. As discussed over [slack thread](https://fueled.slack.com/archives/C077DB69US2/p1758790735423859), we need to account for how we need to handle the admin footer, once we have clarity on it we can work on clipping the screenshot image at viewport limit.

techanvil commented on assets/js/components/setup/SetupUsingProxyWithSignIn/index.js:
```diff
@@ -127,8 +134,13 @@ export default function SetupUsingProxyWithSignIn() {
-				<Grid>
+			<div
+				className={ classnames( 'googlesitekit-setup', {
+					'googlesitekit-setup-content-wrapper':
```
Please use the BEM modifier syntax for this class name, it shouldn't be named as a new top level block.

`--wrapper` won't work so well as a modifier, maybe something like `--with-background` would be more appropriate.


```suggestion
					'googlesitekit-setup-content--with-background':
```

techanvil commented on assets/js/components/setup/SetupUsingProxyWithSignIn/LegacySplashContent.js:
```diff
@@ -0,0 +1,158 @@
+/**
+ * Legacy Splash component for LegacySplash.
```
```suggestion
 * LegacySplashContent component.
```

techanvil commented on assets/js/components/setup/SetupUsingProxyWithSignIn/SplashContent.js:
```diff
@@ -0,0 +1,187 @@
+/**
+ * Refreshed Splash component.
```
```suggestion
 * SplashContent component.
```

techanvil commented on assets/js/components/setup/SetupUsingProxyWithSignIn/index.test.js:
```diff
@@ -100,10 +101,74 @@ describe( 'SetupUsingProxyWithSignIn', () => {
 		).not.toBeInTheDocument();
 	} );
+
+	describe( 'Refresh setup flow', () => {
```
```suggestion
	describe( 'with the `setupFlowRefresh` feature flag enabled', () => {
```

techanvil commented on assets/js/components/setup/SetupUsingProxyWithSignIn/index.test.js:
```diff
@@ -100,10 +101,74 @@ describe( 'SetupUsingProxyWithSignIn', () => {
+		beforeAll( () => {
+			enabledFeatures.add( 'setupFlowRefresh' );
+		} );
+
```
Please add a simple rendering test that includes a snapshot.

techanvil commented on assets/js/components/setup/SetupUsingProxyWithSignIn/Splash.js:
```diff
@@ -24,31 +24,28 @@ import punycode from 'punycode';
-import CompatibilityChecks from '@/js/components/setup/CompatibilityChecks';
+import { useFeature } from '@/js/hooks/useFeature';
+import LegacySplashContent from '@/js/components/setup/SetupUsingProxyWithSignIn/LegacySplashContent';
+import RefreshedSplashContent from '@/js/components/setup/SetupUsingProxyWithSignIn/SplashContent';
```
Let's just call this `SplashContent`, we don't need to prefix it with `Refreshed`.


```suggestion
import SplashContent from '@/js/components/setup/SetupUsingProxyWithSignIn/SplashContent';
```

techanvil reviewed (changes requested):
Thanks for the update @ankitrox, however this one needs another iteration. Please see my comments.

ankitrox commented:
Thank you @techanvil for reviewing the changes. I have made the updates as suggested.

ankitrox commented on assets/js/components/setup/SetupUsingProxyWithSignIn/index.test.js:
```diff
@@ -100,10 +101,74 @@ describe( 'SetupUsingProxyWithSignIn', () => {
 
 		await waitForRegistry();
 
-		expect( container ).toMatchSnapshot();
-
```
Sorry, I think it was removed as part of testing the changes.
Added it again.

ankitrox commented on assets/sass/components/setup/_googlesitekit-splash.scss:
```diff
@@ -0,0 +1,117 @@
+			}
+		}
+
+		.googlesitekit-setup__analytics-opt-in-wrapper {
```
<img width="1163" height="616" alt="Screenshot 2025-09-30 at 10 38 47 AM" src="https://github.com/user-attachments/assets/de6c79fc-4cb5-42b7-9fa4-ac55319cb94b" />

Updated the styles.

ankitrox commented:
<img width="1193" height="635" alt="Screenshot 2025-09-30 at 10 40 49 AM" src="https://github.com/user-attachments/assets/8058f1f2-2011-47fc-b17f-46a47f4ca208" />

Thanks @techanvil

I've addressed other points as mentioned and following as well.

1. Aligned the checkboxes vertically as shown in screenshot.
2. The divider line extended to the edge of the content.
3. There should not be a gap between the green background and the edge of the viewport.
4. For the gap between green background and right hand side of the viewport, I think this will be addressed when we address the screenshot clipping and hiding of admin footer because currently that gap is showing because of the vertical scrollbar.

techanvil reviewed (changes requested):
Thanks @ankitrox.

> 4. For the gap between green background and right hand side of the viewport, I think this will be addressed when we address the screenshot clipping and hiding of admin footer because currently that gap is showing because of the vertical scrollbar.

From what I'm seeing, the gap isn't appearing because of the scrollbar - here's a screenshot of how it looks in a viewport with no scrollbar, the gap still appears.

<img width="1726" height="1281" alt="Image" src="https://github.com/user-attachments/assets/b9ea6d8f-885b-4ca1-8b99-ab8d7533d887" />

However, as mentioned in one of my comments, we need to update this to show gutters around the grid, and this will resolve this gap, so we don't need to worry about it.

ankitrox commented:
<img width="670" height="450" alt="Screenshot 2025-09-30 at 10 28 19 PM" src="https://github.com/user-attachments/assets/41f4f040-1efa-4cf8-99bc-13be11f47e4f" />

Thanks @techanvil 

I've addressed the points which includes.

1. Adding the gutter space for the container.
2. Aligning the opt-in checkbox correctly.
3. Space behing green svg removed (automatically after gutter space).

techanvil reviewed (changes requested):
Thanks @ankitrox!

There are what I'm hoping are the final few issues to address after which this should be good to go. Please see my comments.

Additionally, I've noticed that at viewport widths 1281-1455px, the green background graphic is not visible at all at the right hand edge of the pane. Ideally, we should ensure there's a bit of green between the screenshot and the edge of the pane for all viewport widths, but as we haven't actually got a responsive design defined yet, I think it's OK to leave this as it is for now.

However, we should ensure the bottom right corner of the screenshot is clipped, at the moment it overflows the curved border of the container.

<img width="1300" height="699" alt="Image" src="https://github.com/user-attachments/assets/b9cb9c14-f6b4-4379-a1ca-992cc4d45b83" />

We can achieve this by adding the `overflow: hidden` rule to `.googlesitekit-setup .googlesitekit-layout` in `_googlesitekit-setup.scss`. This looks safe to me. In fact I suspect we can apply it globally to `.googlesitekit-layout`, but as `Layout` is used in quite a few places it's probably better to be cautious and constrain it to the `.googlesitekit-setup` class for now.

```css
	.googlesitekit-setup {

		.googlesitekit-layout {
			overflow: hidden;
		}
```

I've also noticed the output of the "Default - Staged environment warning" story is identical to "Default", however this is also the case for the current splash screen story so I'll create a separate issue to fix both stories and we can leave it as-is for this PR.

techanvil reviewed (changes requested):
Thanks @ankitrox.

There are however still some fixes needed, and with this already significantly over the estimate I've pushed changes to address most of them myself to save time.

I've fixed the grid gutters and the Analytics checkbox alignment, simplified the screenshot rendering and removed an unnecessary version of it, and tweaked the styling for the screenshot to better align with the design.

Please review my changes, and address the last remaining [comment](https://github.com/google/site-kit-wp/pull/11492#discussion_r2410465668) on the PR.

ankitrox commented:
Thanks @techanvil for committing the changes. I've made the updates as requested, also updated the VRT images.

techanvil commented on assets/js/components/setup/SetupUsingProxyWithSignIn/index-setupFlowRefresh.stories.js:
```diff
@@ -0,0 +1,288 @@
+	'Default - Staging environment warning';
+DefaultWithStagingEnvironmentWarning.args = {
+	setupRegistry: ( registry ) => {
+		provideSiteConnection( registry, {
+			hasConnectedAdmins: false,
+			resettable: false,
+		} );
+
```
This call to `provideSiteConnection()` should be removed in order to show the warning notice for this story.

```suggestion
```

techanvil commented on tests/backstop/reference/google-site-kit_Setup___Using_Proxy_With_Sign-in_and_setupFlowRefresh_enabled_Default_0_document_0_small.png:
The tracking checkbox is missing on this reference image, and it's causing its VRT test to [fail](https://github.com/google/site-kit-wp/actions/runs/18342902181/job/52242423957?pr=11492#step:6:7728).
